test/skate_web/controllers/detours_controller_test.exs

```diff
@@ -223,6 +223,66 @@ defmodule SkateWeb.DetoursControllerTest do
                }
              } = json_response(conn, 200)
     end
+
+    @tag :authenticated
+    test "log an error if the serialized detour does not match db state", %{conn: conn} do
+      detour_id = 4
+
+      snapshot = %{
+        "context" => %{
+          "routePattern" => %{
+            "headsign" => "Headsign",
+            "directionId" => 0,
+            "route" => %{
+              "name" => "23"
+            }
+          },
+          "nearestIntersection" => "Street A & Avenue B",
+          "uuid" => detour_id
+        }
+      }
+
+      put(conn, "/api/detours/update_snapshot", %{"snapshot" => snapshot})
+
+      log =
+        CaptureLog.capture_log(fn ->
+          get(conn, "/api/detours/#{detour_id}")
+        end)
+
+      assert log =~
+               "Serialized detour doesn't match saved snapshot. Falling back to snapshot for detour id: #{detour_id}"
+    end
+
+    @tag :authenticated
+    test "fallback to snapshot if the serialized detour does not match db state", %{conn: conn} do
+      detour_id = 4
+
+      snapshot = %{
+        "context" => %{
+          "routePattern" => %{
+            "headsign" => "Headsign",
+            "directionId" => 0,
+            "route" => %{
+              "name" => "23"
+            }
+          },
+          "nearestIntersection" => "Street A & Avenue B",
+          "uuid" => detour_id
+        }
+      }
+
+      put(conn, "/api/detours/update_snapshot", %{"snapshot" => snapshot})
+
+      conn = get(conn, "/api/detours/#{detour_id}")
+
+      assert %{
+               "data" => %{
+                 "author" => "test_user@test.com",
+                 "state" => ^snapshot,
+                 "updated_at" => _
+               }
+             } = json_response(conn, 200)
+    end
```

Comment:
issue: I'm noticing that these tests are only _happening_ to fail because the snapshot is a partial snapshot that doesn't include some of the fields that the serializer adds.

Maybe we should use the `build(:detour)` factory to generate snapshots and update that factory to add the extra required snapshot fields like `children`, `history`, `status`, and `value`?

And then to make these tests fail, we could insert a snapshot into the DB, but then change the `uuid` in the snapshot so that the snapshot `context.uuid` doesn't match the `Detour.id`, which would fail the equivalency check.

This feedback is fairly late, so happy to pair on this if it'd help.

Reply:
For now, we may be able to skip it if it was important to merge this sooner than later, but I think we'd need to do it anyway?

Reply:
Should be fixed now!